components/Tabs/Tabs.js

```diff
@@ -1,38 +1,39 @@
 class TabLink {
   constructor(element){
     // assign this.element to the element reference
-    this.element;
+    this.element = element;
+    console.log(this.element);
     // Get the tab data attribute and save the value here
-    this.tabData = ; 
+    this.tabData = this.element.dataset.tab; 
     // Find all elements with the .card class in index.html that correspond to the tab data attribute
     // If the data is 'all' then select all cards regardless of their data attribute
-    if(){
-      this.cards = ;
+    if(this.tabData === "all"){
+      this.cards = document.querySelectorAll(".card");
     } else {
-      this.cards = ;
+      this.cards = document.querySelectorAll(`.card[data-tab="${this.tabData}"]`);
     }
 
     // Map over the cards array and convert each card element into a new instance of the TabCard class. 
     // Pass in the card object to the TabCard class.
-    this.cards = Array.from(this.cards).map();
+    this.cards = Array.from(this.cards).map(card => {new TabCard(card);});
     // Add a click event that invokes selectTab
-    this.element.addEventListener();
+    this.element.addEventListener("click", this.selectTab());
   }
 
   selectTab(){
 
     // Select all elements with the .tab class on them
-    const tabs = document.querySelectorAll();
+    const tabs = document.querySelectorAll(".tab");
     // Iterate through the NodeList removing the .active-tab class from each element
-    tabs.forEach()
+    tabs.forEach(tabs.classList.remove("active-tab"));
```

Comment:
Your forEach syntax here is missing the parameter and callback/arrow function. It should look something like:

> tabs.forEach( ? => ?);